test/fail_compilation/fail19754.d

```diff
@@ -0,0 +1,20 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail19754.d(13): Error: `cast(int)x` is not an lvalue and cannot be modified
+fail_compilation/fail19754.d(16): Error: `cast(int)x1` is not an lvalue and cannot be modified
+fail_compilation/fail19754.d(19): Error: `cast(int)x2` is not an lvalue and cannot be modified
+---
+*/
+
+void main()
+{
+    shared int x;
+    (cast() x) = 5;
+
+    const int x1;
+    (cast() x1) = 5;
+
+    immutable int x2;
+    (cast() x2) = 5;
```

Comment:
Doesn't this test contradict the spec? These expressions are lvalues: "cast() and cast(qualifier list) when applied to an lvalue."

Reply:
I first wanted to see what happens is shared is treated as const/immutable when it comes to casts

Reply:
Now I implemented it according to the spec